src/app/workflow-editor/object-inspector/io-inspector/workflow-io-inspector.component.ts

```diff
@@ -87,6 +87,41 @@ import {Workflow} from "cwl-svg";
                               [create]="true"></ct-auto-complete>
         </div>
 
+        <!--Batch group-->
+        <div class="form-group" *ngIf="workflowModel.hasBatch && isInputPort()">
+
+            <label class="form-control-label">Create batch group
+
+                <span class="text-warning small" *ngIf="customValueIsNotSelected()">
+                      <i class="fa fa-warning fa-fw" [ct-tooltip]="ctt"></i>
+                </span>
+ 
+                <ct-tooltip-content #ctt>
+                    <div class="tooltip-info">
+                        This workflow has a batch criteria which has been set via the API. If you change it,
+                        you won't be able to restore the custom batch criteria afterwards.
+                    </div>
+                </ct-tooltip-content>
+
+            </label>
+
+            <!--Batch select-->
+            <select class="form-control" *ngIf="!workflowModel['batchInput'] || workflowModel['batchInput'] === port.id"
+                    [formControl]="form.controls['batchType']">
+                <option *ngFor="let propertyType of batchByList" [ngValue]="propertyType.value">
+                    {{propertyType.label}}
+                </option>
+            </select>
+
+            <!--Warning when some other input is already configured as batch-->
+            <div class="text-warning small"
+                 *ngIf="workflowModel['batchInput'] && workflowModel['batchInput'] !== port.id">
```

Comment:
Use dot notation `workflowModel.batchInput`, your linter will complain if this field ever gets changed.